frontend/ui/widget/bookmapwidget.lua

```diff
@@ -1939,6 +1936,51 @@ function BookMapWidget:onTap(arg, ges)
     return true
 end
 
+function BookMapWidget:onGoToFocusedPage()
+    if not self.enable_focus_navigation or not self.cur_focused_widget then
+        return true
+    end
+
+    local target_page = nil
+    -- Find the row containing the focused widget.
+    local row = self:getVGroupRowAtY(self.cur_focused_widget.dimen.y - self.title_bar_h)
+    if row and row.start_page then
+        -- Find the focused widget in the row's layout.
+        local invisible_page_slots = row.focus_layout[#row.focus_layout] -- last row in focus_layout contains page slots
+        if invisible_page_slots then
+            for i, widget in ipairs(invisible_page_slots) do
+                if widget == self.cur_focused_widget then
+                    target_page = row.start_page + i - 1
+                    break
+                end
+            end
+        end
+    end
+    if target_page then
+        local should_go_to_page_browser
+        if self.overview_mode then
+            should_go_to_page_browser = G_reader_settings:isFalse("book_map_overview_tap_to_page_browser")
+        else
+            should_go_to_page_browser = G_reader_settings:isFalse("book_map_tap_to_page_browser")
+        end
+
+        if should_go_to_page_browser then
```

Comment:
(nitpick, blank line not needed (both sections are related) - question of taste, but the overall bookmapwidget.lua tastes my taste :) so question of consistency :)